### ISSUE TITLE
[1.10.2] un able to craft and un able to spawn in on server

### ISSUE THREAD

**Issue by audenga** [codetaylor/ct-watercan#2]
title

**Comment by codetaylor** [comment]
Fixed in 0.4.4

Was trying to load client code on server.

Confirmed can craft / can use on 1.12.2 server.

**Comment by audenga** [comment]
using a custom 1.10.2 pack through ftb/curse/twitch or what ever it wants to be called nowdays

**Comment by codetaylor** [comment]
Ah, so this is for the 1.10.2 version? My mistake.

**Comment by codetaylor** [comment]
Confirmed in 1.10.2

Caused by: java.lang.RuntimeException: Attempted to load class net/minecraft/client/entity/AbstractClientPlayer for invalid side SERVER

---------------------------------------------

```
[12:03:41] [Server thread/ERROR] [KorEventService]: Unable to invoke method [public void com.sudoplay.mc.korwatercan.module.watercan.ModuleWaterCan.onRegisterItemsEvent(com.sudoplay.mc.kor.spi.event.internal.OnRegisterItemsEvent)] with event object [com.sudoplay.mc.kor.spi.event.internal.OnRegisterItemsEvent@d3b824c]
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
	at com.sudoplay.mc.kor.core.event.service.EventDelegate.call(EventDelegate.java:22) ~[EventDelegate.class:?]
	at com.sudoplay.mc.kor.core.event.service.EventService.publish(EventService.java:69) [EventService.class:?]
	at com.sudoplay.mc.kor.spi.Kor.onPreInitialization(Kor.java:162) [Kor.class:?]
	at com.sudoplay.mc.korwatercan.KorWaterCan.onPreInitialization(KorWaterCan.java:59) [KorWaterCan.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:602) [FMLModContainer.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74) [guava-17.0.jar:?]
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47) [guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322) [guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304) [guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:275) [guava-17.0.jar:?]
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:243) [LoadController.class:?]
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:221) [LoadController.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_144]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_144]
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74) [guava-17.0.jar:?]
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47) [guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322) [guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304) [guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:275) [guava-17.0.jar:?]
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:145) [LoadController.class:?]
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:614) [Loader.class:?]
	at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:98) [FMLServerHandler.class:?]
	at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:328) [FMLCommonHandler.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.startServer(DedicatedServer.java:124) [DedicatedServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:497) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_144]
Caused by: java.lang.RuntimeException: Failed to instantiate class com.sudoplay.mc.korwatercan.module.watercan.item.ItemWaterCanWood
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:84) ~[RegistryService.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:58) ~[RegistryService.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.ModuleWaterCan.onRegisterItemsEvent(ModuleWaterCan.java:60) ~[ModuleWaterCan.class:?]
	... 40 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_144]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_144]
	at com.sudoplay.mc.kor.core.registry.service.injection.strategy.constructor.ArgConstructorStrategy.instantiate(ArgConstructorStrategy.java:72) ~[ArgConstructorStrategy.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.instantiate(RegistryObjectInjector.java:44) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.createInjectedObject(RegistryObjectInjector.java:29) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:81) ~[RegistryService.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:58) ~[RegistryService.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.ModuleWaterCan.onRegisterItemsEvent(ModuleWaterCan.java:60) ~[ModuleWaterCan.class:?]
	... 40 more
Caused by: java.lang.NoClassDefFoundError: net/minecraft/client/entity/AbstractClientPlayer
	at com.sudoplay.mc.korwatercan.module.watercan.item.AbstractItemWaterCan.<init>(AbstractItemWaterCan.java:81) ~[AbstractItemWaterCan.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.item.ItemWaterCanWood.<init>(ItemWaterCanWood.java:46) ~[ItemWaterCanWood.class:?]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_144]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_144]
	at com.sudoplay.mc.kor.core.registry.service.injection.strategy.constructor.ArgConstructorStrategy.instantiate(ArgConstructorStrategy.java:72) ~[ArgConstructorStrategy.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.instantiate(RegistryObjectInjector.java:44) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.createInjectedObject(RegistryObjectInjector.java:29) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:81) ~[RegistryService.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:58) ~[RegistryService.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.ModuleWaterCan.onRegisterItemsEvent(ModuleWaterCan.java:60) ~[ModuleWaterCan.class:?]
	... 40 more
Caused by: java.lang.ClassNotFoundException: net.minecraft.client.entity.AbstractClientPlayer
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191) ~[launchwrapper-1.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[?:1.8.0_144]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_144]
	at com.sudoplay.mc.korwatercan.module.watercan.item.AbstractItemWaterCan.<init>(AbstractItemWaterCan.java:81) ~[AbstractItemWaterCan.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.item.ItemWaterCanWood.<init>(ItemWaterCanWood.java:46) ~[ItemWaterCanWood.class:?]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_144]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_144]
	at com.sudoplay.mc.kor.core.registry.service.injection.strategy.constructor.ArgConstructorStrategy.instantiate(ArgConstructorStrategy.java:72) ~[ArgConstructorStrategy.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.instantiate(RegistryObjectInjector.java:44) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.createInjectedObject(RegistryObjectInjector.java:29) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:81) ~[RegistryService.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:58) ~[RegistryService.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.ModuleWaterCan.onRegisterItemsEvent(ModuleWaterCan.java:60) ~[ModuleWaterCan.class:?]
	... 40 more
Caused by: net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerException: Exception in class transformer net.minecraftforge.fml.common.asm.transformers.SideTransformer@6928f576 from coremod FMLCorePlugin
	at net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerWrapper.transform(ASMTransformerWrapper.java:253) ~[forgeSrc-1.10.2-12.18.2.2151.jar:?]
	at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:279) ~[launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:176) ~[launchwrapper-1.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[?:1.8.0_144]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_144]
	at com.sudoplay.mc.korwatercan.module.watercan.item.AbstractItemWaterCan.<init>(AbstractItemWaterCan.java:81) ~[AbstractItemWaterCan.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.item.ItemWaterCanWood.<init>(ItemWaterCanWood.java:46) ~[ItemWaterCanWood.class:?]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_144]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_144]
	at com.sudoplay.mc.kor.core.registry.service.injection.strategy.constructor.ArgConstructorStrategy.instantiate(ArgConstructorStrategy.java:72) ~[ArgConstructorStrategy.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.instantiate(RegistryObjectInjector.java:44) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.createInjectedObject(RegistryObjectInjector.java:29) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:81) ~[RegistryService.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:58) ~[RegistryService.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.ModuleWaterCan.onRegisterItemsEvent(ModuleWaterCan.java:60) ~[ModuleWaterCan.class:?]
	... 40 more
Caused by: java.lang.RuntimeException: Attempted to load class net/minecraft/client/entity/AbstractClientPlayer for invalid side SERVER
	at net.minecraftforge.fml.common.asm.transformers.SideTransformer.transform(SideTransformer.java:56) ~[forgeSrc-1.10.2-12.18.2.2151.jar:?]
	at net.minecraftforge.fml.common.asm.ASMTransformerWrapper$TransformerWrapper.transform(ASMTransformerWrapper.java:249) ~[forgeSrc-1.10.2-12.18.2.2151.jar:?]
	at net.minecraft.launchwrapper.LaunchClassLoader.runTransformers(LaunchClassLoader.java:279) ~[launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:176) ~[launchwrapper-1.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[?:1.8.0_144]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[?:1.8.0_144]
	at com.sudoplay.mc.korwatercan.module.watercan.item.AbstractItemWaterCan.<init>(AbstractItemWaterCan.java:81) ~[AbstractItemWaterCan.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.item.ItemWaterCanWood.<init>(ItemWaterCanWood.java:46) ~[ItemWaterCanWood.class:?]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_144]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_144]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_144]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_144]
	at com.sudoplay.mc.kor.core.registry.service.injection.strategy.constructor.ArgConstructorStrategy.instantiate(ArgConstructorStrategy.java:72) ~[ArgConstructorStrategy.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.instantiate(RegistryObjectInjector.java:44) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.injection.RegistryObjectInjector.createInjectedObject(RegistryObjectInjector.java:29) ~[RegistryObjectInjector.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:81) ~[RegistryService.class:?]
	at com.sudoplay.mc.kor.core.registry.service.RegistryService.register(RegistryService.java:58) ~[RegistryService.class:?]
	at com.sudoplay.mc.korwatercan.module.watercan.ModuleWaterCan.onRegisterItemsEvent(ModuleWaterCan.java:60) ~[ModuleWaterCan.class:?]
	... 40 more
```

**Comment by codetaylor** [comment]
Fixed in version 1.10.2-0.4.1, waiting on CurseForge approval.

Thank you for finding this issue and please let me know if this latest 1.10.2 build solves it. :)

**Comment by audenga** [comment]
confirmed working
